resource_leak.py

```diff
@@ -0,0 +1,44 @@
+# DB resource leak demo
+import psycopg2
+
+def test_connection():
+    connection = psycopg2.connect(user="postgres",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *connection*. The resource is allocated by call *psycopg2.connect*. Execution paths that do not contain closure statements were detected. Close *connection* in a try-finally block to prevent the resource leak.